crates/goose/src/agents/moim.rs

```diff
@@ -0,0 +1,147 @@
+use crate::agents::extension_manager::ExtensionManager;
+use crate::conversation::message::Message;
+use crate::conversation::{fix_conversation, Conversation};
+use rmcp::model::Role;
+
+pub async fn inject_moim(
+    conversation: Conversation,
+    extension_manager: &ExtensionManager,
+) -> Conversation {
+    let config = crate::config::Config::global();
+    if !config.get_param("GOOSE_MOIM_ENABLED").unwrap_or(true) {
+        return conversation;
+    }
+
+    if let Some(moim) = extension_manager.collect_moim().await {
+        let mut messages = conversation.messages().clone();
+        let idx = messages
+            .iter()
+            .rposition(|m| m.role == Role::Assistant)
+            .unwrap_or(messages.len());
+        messages.insert(idx, Message::user().with_text(moim));
+
+        let (fixed, _issues) = fix_conversation(Conversation::new_unvalidated(messages));
+        return fixed;
+    }
```

Comment:
I would suggest here to return the origianl conversation if there are issues. 

and then in the tests, do one that has a bunch of cases, but the cases are all just conversations and verify that the conversation returned are different from what we put in.